crates/transport-http/src/hyper_transport.rs

```diff
@@ -1,74 +1,96 @@
-use crate::{Http, HttpConnect};
 use alloy_json_rpc::{RequestPacket, ResponsePacket};
 use alloy_transport::{
     utils::guess_local_url, TransportConnect, TransportError, TransportErrorKind, TransportFut,
 };
 use http_body_util::{BodyExt, Full};
 use hyper::{
-    body::{Buf, Bytes},
-    header,
+    body::{Bytes, Incoming},
+    header, Request, Response,
 };
-use hyper_util::client::legacy::{connect::Connect, Client};
-use std::task;
+use hyper_util::client::legacy::Error;
+use std::{future::Future, marker::PhantomData, pin::Pin, task};
 use tower::Service;
 use tracing::{debug, debug_span, trace, Instrument};
+use url::Url;
+
+use crate::{Http, HttpConnect};
 
 /// A [`hyper`] HTTP client.
 pub type HyperClient = hyper_util::client::legacy::Client<
     hyper_util::client::legacy::connect::HttpConnector,
     http_body_util::Full<::hyper::body::Bytes>,
 >;
 
-/// An [`Http`] transport using [`hyper`].
-pub type HyperTransport = Http<HyperClient>;
+/// A [hyper] client that can be used with tower layers.
+#[derive(Clone, Debug)]
+pub struct HyperTransport<B = Full<Bytes>, S = HyperClient> {
+    service: S,
+    _pd: PhantomData<B>,
+}
 
-/// Connection details for a [`HyperTransport`].
-pub type HyperConnect = HttpConnect<HyperTransport>;
+/// Alias for [`Response<Incoming>`]
+pub type HyperResponse = Response<Incoming>;
 
-impl TransportConnect for HyperConnect {
-    type Transport = HyperTransport;
+/// Alias for pinned box future that results in [`HyperResponse`]
+pub type HyperResponseFut<T = HyperResponse, E = Error> =
+    Pin<Box<dyn Future<Output = Result<T, E>> + Send + 'static>>;
 
-    fn is_local(&self) -> bool {
-        guess_local_url(self.url.as_str())
-    }
-
-    fn get_transport<'a: 'b, 'b>(
-        &'a self,
-    ) -> alloy_transport::Pbf<'b, Self::Transport, TransportError> {
+impl HyperTransport {
+    /// Create a new [HyperTransport] with the given URL and default hyper client.
+    pub fn new() -> Self {
         let executor = hyper_util::rt::TokioExecutor::new();
 
-        let client = hyper_util::client::legacy::Client::builder(executor).build_http();
+        let service =
+            hyper_util::client::legacy::Client::builder(executor).build_http::<Full<Bytes>>();
+
+        Self { service, _pd: PhantomData }
+    }
+}
 
-        Box::pin(async move { Ok(Http::with_client(client, self.url.clone())) })
+impl Default for HyperTransport {
+    fn default() -> Self {
+        Self::new()
     }
 }
 
-impl<C, B> Http<Client<C, Full<B>>>
+impl<B, S> HyperTransport<B, S> {
+    /// Create a new [HyperTransport] with the given URL and service.
+    pub const fn with_service(service: S) -> Self {
+        Self { service, _pd: PhantomData }
+    }
+}
+
+impl<B, S> HyperTransport<B, S>
```

Comment:
can we keep this impl block identical to the one we had before? previous implementation was for `Http<...>`, so we didn't have to pass `url` arg, there's no reason to change that